roles/openshift_health_checker/openshift_checks/memory_availability.py

```diff
@@ -0,0 +1,33 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class MemoryAvailability(OpenShiftCheck):
+    """Check that recommended memory is available."""
+
+    name = "memory_availability"
+    tags = ["preflight"]
+
+    recommended_memory_gb = {
+        "nodes": 8.0,
+        "masters": 16.0,
+        "etcd": 20.0
+    }
+
+    def run(self, tmp, task_vars):
+        group_names = get_var(task_vars, "group_names", default=[])
+        memoryfree = get_var(task_vars, "ansible_memtotal_mb")
+
+        recommended_memory_gb = max(self.recommended_memory_gb.get(group, 0) for group in group_names)
+
+        if self.to_gigabytes(memoryfree) < recommended_memory_gb:
```

Comment:
Why don't we store the recommended values in the unit we want to compare to (MB)?